### PR TITLE
APP-474: Add topic name for received notifications

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @Slash32

### CODE REVIEW

**Comment by Slash32** [comment]
@kaa-leeroy test this please

File: notificationdemo/source/c/src/kaa_demo.c [hunk 1]
```diff
@@ -117,17 +117,17 @@ static void process_user_command()
 static void on_notification(void *context, uint64_t *topic_id, kaa_notification_t *notification)
 {
     (void)context;
+    kaa_list_node_t *t = kaa_list_find_next(kaa_list_begin(topic_list), find_topic_predicate, topic_id);
```
Comment (ohamov) [inline notificationdemo/source/c/src/kaa_demo.c:120]: Check search result

**Review by ohamov — CHANGES_REQUESTED** [review]

File: notificationdemo/source/c/src/kaa_demo.c (hunk 1 above)
Comment (RostakaGmfun) [reply notificationdemo/source/c/src/kaa_demo.c:120]: Should we expect notifications for topics we are not subscribed to?

File: notificationdemo/source/c/src/kaa_demo.c [hunk 2]
```diff
@@ -117,17 +117,20 @@ static void process_user_command()
...
+        demo_printf("Notification for topic id '%llu', name '%s' received\r\n", *topic_id, topic->name);
         demo_printf("Notification body: %s\r\n", message->data);
         demo_printf("Message alert type: %s\r\n", color_to_str(notification->alert_type));
     } else {
         demo_printf("Error: Received notification's body is null\r\n");
     }
 
-    kaa_list_node_t *t = kaa_list_find_next(kaa_list_begin(topic_list), find_topic_predicate, topic_id);
-    kaa_topic_t *topic = kaa_list_get_data(t);
     if (topic && topic->subscription_type == OPTIONAL_SUBSCRIPTION) {
```
Comment (rasendubi) [inline notificationdemo/source/c/src/kaa_demo.c]: `topic` was dereferenced before checking for `NULL`. (Dereferenced in line 127.)

**Comment by Slash32** [comment]
@kaa-leeroy test this please

**Comment by Slash32** [comment]
@kaa-leeroy test this please

**Comment by Slash32** [comment]
@kaa-leeroy test this please

**Review by Slash32 — APPROVED** [review]

**Pull request merged**